lib/omniauth/strategies/openid_connect.rb

```diff
@@ -109,7 +111,8 @@ def callback_phase
 
         raise CallbackError, 'Invalid state parameter' if invalid_state
 
-        return fail!(:missing_code, OmniAuth::OpenIDConnect::MissingCodeError.new(params['error'])) unless params['code']
+        grant_type_errors = validate_grant_type_credentials
+        return fail!(*grant_type_errors) if grant_type_errors.present?
```

Comment:
Sorry, I was wrong about raising just first exception. It looks like this code will fail if array has more than one exception.

Reply:
This code will actually work, the splat operator simply extracts only two elements of the returned array: message_key and exception. You're right that this code will only return the first exception.
So the question is, if the user (programmer?) specified "id_token foo" as the answer type, should we still try to work with `id_token` (valid) and ignore foo (invalid)? Or should we return an error indicating that this response_type is incorrect as a whole (that was my assumption).

Reply:
Well, since we support only two types `id_token` & `code` I would treat all other values as invalid and show exception as soon as possible.

Reply:
That's what this code is doing, it's returing one exception as soon as it finds it.

But as I mentioned earlier, I think that we need to take a step back, remove support for the `token` response type and focus on `id_token` only.

Reply:
I mean there's no need to parse `type` value, try to split it to several values and so far. Just compare it directly, and if there's no match - raise an error.